pkg/controller/plan/kubevirt.go

```diff
@@ -766,42 +767,22 @@ func (r *KubeVirt) areOvirtPVCsReady(vm ref.Ref, step *plan.Step) (ready bool, e
 	return
 }
 
-var filesystemMode = core.PersistentVolumeFilesystem
-
-// Return storage profile access mode based on storage class
-func (r *KubeVirt) getStorageProfileModes(storageName string) (accessModes []core.PersistentVolumeAccessMode, volumeMode *core.PersistentVolumeMode, err error) {
-	storageProfileList := &cdi.StorageProfileList{}
-	err = r.Client.List(context.TODO(), storageProfileList)
+func (r *KubeVirt) getDefaultVolumeAndAccessMode(storageName string) ([]core.PersistentVolumeAccessMode, *core.PersistentVolumeMode, error) {
+	storageProfile := &cdi.StorageProfile{}
+	err := r.Client.Get(context.TODO(), types.NamespacedName{Name: storageName}, storageProfile)
 	if err != nil {
-		return
+		return nil, nil, liberr.Wrap(err, "cannot get StorageProfile")
 	}
-	for i := range storageProfileList.Items {
-		storageProfile := &storageProfileList.Items[i]
-		if storageProfile.Name == storageName {
-			for _, claimProperty := range storageProfile.Status.ClaimPropertySets {
-				volumeMode = claimProperty.VolumeMode
-				if volumeMode != nil && *volumeMode == core.PersistentVolumeBlock {
-					// Preferring Block volume mode
-					break
-				}
-			}
-			if volumeMode == nil {
-				// volumeMode is an optional API parameter. Filesystem is the default mode used when volumeMode parameter is omitted.
-				volumeMode = &filesystemMode
-			}
-			for _, claimProperty := range storageProfile.Status.ClaimPropertySets {
-				claimPropertyVolumeMode := claimProperty.VolumeMode
-				if claimPropertyVolumeMode == nil {
-					claimPropertyVolumeMode = &filesystemMode
-				}
-				if *claimPropertyVolumeMode == *volumeMode {
-					accessModes = append(accessModes, claimProperty.AccessModes...)
-				}
-			}
-			break
-		}
+
+	if len(storageProfile.Status.ClaimPropertySets) > 0 &&
+		len(storageProfile.Status.ClaimPropertySets[0].AccessModes) > 0 {
+		accessModes := storageProfile.Status.ClaimPropertySets[0].AccessModes
+		volumeMode := storageProfile.Status.ClaimPropertySets[0].VolumeMode
```

Comment:
AFAIR, we failed in this place before when volume mode was missing (and thus, defaulted to Filesystem) but just to be sure that we don't bring up this issue again (and if we do, we'll need to change the place we fail on), can you please check what happens when `len(storageProfile.Status.ClaimPropertySets[0].AccessModes) > 0` and volumeMode is not specified?

Reply:
yup, good catch, good that we didn't introduce a problem we already hit.